src/android/DatagramSocketListener.java

```diff
@@ -0,0 +1,79 @@
+package org.apache.cordova.dgram;
+
+import android.util.Log;
+
+import org.apache.cordova.CallbackContext;
+import org.apache.cordova.PluginResult;
+
+import org.json.JSONObject;
+
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+
+import java.nio.charset.StandardCharsets;
+
+class DatagramSocketListener extends Thread {
+    private DatagramSocket datagramSocket;
+    private CallbackContext callbackContext;
+
+    DatagramSocketListener(
+        final DatagramSocket datagramSocket,
```

Comment:
This is atypical formatting for a java project, not sure how its happening.

Reply:
what do you think of using something like this?  https://marketplace.visualstudio.com/items?itemName=redhat.java